Feat: GameRecord 관련 파일 생성 및 수정

skybluelion opened the pull request:
## 📌 관련 이슈
Feat: 토론 게임 전적 관련 기능 구현 #11

## ✨ 과제 내용
토론 게임 전적 관련 기능 구현
 - 전적 생성
 - 아이디별 전적 조회
 - 나와 상대의 전적 조회
 - 전체 랭킹 조회
 - 게임 후 MMR 갱신

## 📸 스크린샷
 - 전적 생성(수정)
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/895b31a1-e6a9-477c-a484-4fe7bbd616fc)

![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/4aabac8b-cfd4-4d29-a5f7-c86db8f5dda1)

 - 아이디별 전적 조회
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/a6f71976-93be-476d-bb8b-1fe489493a29)


 - 나와 상대의 전적 조회
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/339158bd-8f99-4bc3-a597-85df559bc3fb)


 - 전체 랭킹 조회(수정)
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/545a6d0e-db47-4c4c-8a31-ef47e1466948)


 - 게임 후 MMR 갱신
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/8e2e04aa-5aae-4673-93dd-089e8bef012e)
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/a40a1f51-f313-4b74-9ba4-7917653e7d9b)
![image](https://github.com/GoLAB-Project/GoLAB-backend/assets/99467446/adb78e63-eef8-47eb-85e6-9c66e6161994)


23.08.27 코드 및 캡쳐 수정 완료


## 📚 레퍼런스 (또는 새로 알게 된 내용) 혹은 궁금한 사항들
alias

skybluelion committed:
[Feat] GameRecord 관련 파일 생성 및 수정
- GameRecordDto, GameRecordWithRankDto, PlayerDto, GameRecordrepository, GameRecordservice, GameRecordserviceImpl, GameRecordController, GameRecord 수정

mukhoplus reviewed (commented):
1. 저도 다 지킨 것 같지는 않으나, 컨트롤러 부분에서 한번 확인해보시면 좋을 것 같습니다.

https://dev-cool.tistory.com/32

2. 나와 상대의 전적 조회가 '상대 전적 조회'가 아니라 '내 전적'과 '상대 전적'을 조회하는 로직인가요?

skybluelion committed:
[Feat] GameRecord 관련 수정
- GameRecordrepository, GameRecordservice,GameRecordserviceImpl, GameRecordController 수정

skybluelion commented:
> 1. 저도 다 지킨 것 같지는 않으나, 컨트롤러 부분에서 한번 확인해보시면 좋을 것 같습니다.
> https://dev-cool.tistory.com/32
> 확인 후 수정했습니다!
> 2. 나와 상대의 전적 조회가 '상대 전적 조회'가 아니라 '내 전적'과 '상대 전적'을 조회하는 로직인가요?
> 넵 맞습니다. 게임 시작 시 혹은 게임 종료 후 뜰 두 플레이어의 전적 모두 검색하는 것입니다.

mukhoplus reviewed (approved):
👍

richsubin reviewed (approved):
고생하셨습니다 👍👍
